### PR TITLE
Pluggable loggers including FileLogger

### PR DESCRIPTION
**Pull request by nowsprinting**
### Changes
- Change logger settings to pluggable
- Refactor and add filtering LogType option into ConsoleLogger
- Add CompositeLogger
- Add FileLogger

### Discussions
Can I change `Autopilot.CreateDefaultLogger` method accessor to private?

---

### Contribution License Agreement

- [x] By placing an "x" in the box, I hereby understand, accept and agree to be bound by the terms and conditions of the [Contribution License Agreement](https://dena.github.io/cla/).

### CODE REVIEW

**Comment by nowsprinting** [comment]
I found a bug...

File: Runtime/Loggers/AbstractLogger.cs [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+        /// <summary>
+        /// Description about this agent instance.
+        /// </summary>
+        [Multiline] public string description;
+#endif
+
+        /// <summary>
+        /// Logger implementation used for autopilot.
+        /// </summary>
+        public abstract ILogger LoggerImpl { get; }
```
Comment (Kuniwak) [inline Runtime/Loggers/AbstractLogger.cs:24]: How about this suggestion?

```suggestion
    public abstract class AbstractLoggerSettings : ScriptableObject, IDisposable
    {
#if UNITY_EDITOR
        /// <summary>
        /// Description about this agent instance.
        /// </summary>
        [Multiline] public string description;
#endif

        /// <summary>
        /// Logger implementation used for autopilot.
        /// </summary>
        public abstract ILogger Logger { get; }
```

File: Runtime/Loggers/CompositeLogger.cs [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+    [CreateAssetMenu(fileName = "New CompositeLogger", menuName = "Anjin/Composite Logger", order = 70)]
+    public class CompositeLogger : AbstractLogger
+    {
+        /// <summary>
+        /// Loggers to delegates
+        /// </summary>
+        public List<AbstractLogger> loggers = new List<AbstractLogger>();
+
+        private ILogHandler _handler;
+        private ILogger _logger;
```
[the remark this answers is not in the document]
Comment (nowsprinting) [reply Runtime/Loggers/CompositeLogger.cs]: Thanks!

File: Runtime/Loggers/AbstractLogger.cs (hunk 1 above)
Comment (nowsprinting) [reply Runtime/Loggers/AbstractLogger.cs:24]: Thanks, I'm a little hesitant to name it "settings", but I don't have another idea...

**Comment by nowsprinting** [comment]
@Kuniwak san,

> Can I change Autopilot.CreateDefaultLogger method accessor to private?

**Comment by nowsprinting** [comment]
Re-rename `AbstractLogger` to `AbstractLoggerAsset` (Use "Assets" instead of "Settings").

**Review by Kuniwak — APPROVED** [review]
LGTM! Thank you for the contributions!

**Pull request merged**